src/common/utils/ldap/ldap.go

```diff
@@ -432,6 +432,41 @@ func (session *Session) searchGroup(groupDN, filter, gName, groupNameAttribute s
 	return ldapGroups, nil
 }
 
+// SearchAllGroups -- Fetch all groups matching the filter from LDAP
+func (session *Session) SearchAllGroups() ([]models.LdapGroup, error) {
+	ldapGroups := make([]models.LdapGroup, 0)
+
+	// Search all groups with LDAP group filter condition
+	ldapFilter, err := createGroupSearchFilter(session.ldapGroupConfig.LdapGroupFilter, "", session.ldapGroupConfig.LdapGroupNameAttribute)
+	if err != nil {
+		log.Errorf("wrong filter format: filter:%v, groupNameAttribute:%v", session.ldapGroupConfig.LdapGroupFilter, session.ldapGroupConfig.LdapGroupNameAttribute)
+		return ldapGroups, err
+	}
+
+	// There maybe many groups under the LDAP group base DN
+	// If return all groups in LDAP group base DN, it might get "Size Limit Exceeded" error
+	result, err := session.SearchLdapAttribute(session.groupBaseDN(), ldapFilter, []string{session.ldapGroupConfig.LdapGroupNameAttribute})
```

Comment:
Could you please handle the case when "Size Limit Exceeded" error?

Reply:
I think this case is not possible because the SizeLimit parameter in SearchRequest is 0:
https://github.com/goharbor/harbor/blob/5d9a01152acbbdb7c2d06a88b1f7f3161f07d72a/src/common/utils/ldap/ldap.go#L310

Reply:
Moreover, this error should theoretically be handled in this code:
https://github.com/goharbor/harbor/blob/5d9a01152acbbdb7c2d06a88b1f7f3161f07d72a/src/common/utils/ldap/ldap.go#L318

Reply:
It is possible! The "Size Limit Exceeded" is an error returned by the LDAP Server. This was fixed in v2.1.1, i think, with https://github.com/goharbor/harbor/blob/a211b0c9d79b85368bf8c6c4c340e0ceb2cfc70e/src/common/utils/ldap/ldap.go#L411-L414

Reply:
I'm sorry. You are really right
For such a case, we must implement a request with pagination in LDAP. Page Size should be configured in Authentication page.

@DASXCE
Can you help me implement this?

Reply:
@DASXCE @stonezdj
I implemented pagination for LDAP queries. This should help in case of the "Size Limit Exceeded" error.